pkg/didcomm/transport/transport_interface.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package transport
+
+// OutboundTransport interface definition for transport layer
+// This is the client side of the agent
+type OutboundTransport interface {
+	// Send send a2a exchange data
+	Send(data string, destination string) (string, error)
+}
+
+// RequestRouter struct for path and handler function
+type RequestRouter struct {
+	Path        string
+	HandlerFunc func([]byte) error
+}
+
+// DIDCommHandler struct to pass details for request handlers
+type DIDCommHandler struct {
+	RecieveInvitation    *RequestRouter
```

Comment:
This should not be tightly coupled to specific message types.

Reply:
Inbound handling will be done in a separate PR